src/languages/common.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{
+    lilypond_objects::lilypond_note::LilyPondNote,
+    notation::{
+        note::Note,
+        pitch::Octave,
+        rhythm::{DurationType, Length, Rhythm},
+    },
+};
+
+pub fn octave(note: &Note) -> &str {
+    match note.pitch.octave {
+        Octave::S0 => ",,,",
+        Octave::S1 => ",,",
+        Octave::S2 => ",",
+        Octave::S3 => "",
+        Octave::S4 => "'",
+        Octave::S5 => "''",
+        Octave::S6 => "'''",
+        Octave::S7 => "''''",
+        Octave::S8 => "'''''",
+        Octave::S9 => "''''''",
+        Octave::None => "",
+    }
+}
+
+pub fn length(note: &Note) -> &str {
+    match note.rhythm.length {
+        Length::Whole => "1",
+        Length::Half => "2",
+        Length::Quarter => "4",
+        Length::Eighth => "8",
+        Length::Sixteenth => "16",
+        Length::ThirtySecond => "32",
+        Length::SixtyFourth => "64",
+        Length::OneTwentyEighth => "128",
+    }
+}
+
+pub fn dotted(note: &Note) -> &str {
+    match note.rhythm.dotted {
+        true => ".",
+        false => "",
+    }
+}
+
+pub fn duration_type_from_lilypond(note: &LilyPondNote) -> DurationType {
+    match note.get_capture("note_name").as_str() {
+        "r" => DurationType::Rest,
+        _ => DurationType::Note,
+    }
+}
+
+pub fn octave_from_lilypond(note: &LilyPondNote) -> Octave {
+    match duration_type_from_lilypond(note) {
+        DurationType::Rest => Octave::None,
+        DurationType::Note => {
+            // octave has to be usize to add count() results from it
+            let mut octave_int: usize = 3;
+            let octave_string = note.get_capture("octave");
+
+            if octave_string.contains(",") && octave_string.contains("'") {
+                // Check for both octave transposition characters and panic
+                panic!("Mixed octave transpostion symbols , and '.");
+            } else if octave_string.contains("'") {
+                octave_int += octave_string.matches("'").count();
+            } else if octave_string.contains(",") {
+                octave_int -= octave_string.matches(",").count();
+            }
+
+            match octave_int {
+                0 => Octave::S0,
+                1 => Octave::S1,
+                2 => Octave::S2,
+                3 => Octave::S3,
+                4 => Octave::S4,
+                5 => Octave::S5,
+                6 => Octave::S6,
+                7 => Octave::S7,
+                8 => Octave::S8,
+                9 => Octave::S9,
+                _ => panic!("Invalid number of octave transpositions."),
```

Comment:
The library should use std::result::Result and have an Error type instead of panicking on bad data IMO.

Reply:
Thank you for your feedback! I have just removed every `panic!` call from the library. I haven't yet been able to implement your other feedback, but I should be able to within a few days.